changelog/5023.feature.rst

```diff
@@ -0,0 +1 @@
+``--strict`` is now called ``--strict-markers`` as it is more explicit about what it does. The old name still works for backward compatibility.
```

Comment:
With regard to `--strict` I would keep it anyway, being able to select also other strict modes later..

Reply:
And therefore it could still be used in default config (when the intention here is to enforce strict checkings in general).

Reply:
> With regard to --strict I would keep it anyway, being able to select also other strict modes later..

It has not been removed, I only added `--strict-markers` as the now preferred name, `--strict` is still present and works as before.

Reply:
I know.. it just sounded like it would be only kept for backward compatibility, while I think it makes sense to keep it in general.

Reply:
And `--strict` should be used in some test probably still.

Reply:
Indeed, I thought the same thing, please see `test_strict_prohibits_unregistered_markers`

> I know.. it just sounded like it would be only kept for backward compatibility, while I think it makes sense to keep it in general.

Not sure if we should ever change the meaning of `--strict` in the future as well. I mean, suppose we decide to turn "cannot collect classes with __init__" into an error using `--strict`. If we do that, then test suites which used `--strict` to turn marks into errors will now break.

In the future we probably should grow another option for that kind of check and leave this option alone?

Reply:
@blueyed any thoughts on this? If you feel strongly about changing the meaning of `--strict` and wording, I don't mind changing it back.

Reply:
> If we do that, then test suites which used `--strict` to turn marks into errors will now break.

I'd argue that this is OK - and easy to fix then.
It is easier to have `--strict` for CI etc, than having to add multiple `--strict-*` variants then.
New strictness checks could still get their own option, of course (if it makes sense).